packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestEditorState.ts

```diff
@@ -221,6 +235,94 @@ export class ServiceTestSetupState {
     }));
   }
 
+  getBindingWithParamFromQuery(): {
+    binding: Binding;
+    param: string;
+  }[] {
+    const query =
+      this.testState.suiteState.testableState.serviceEditorState.serviceQuery;
+    if (query && !isStubbed_RawLambda(query)) {
+      // safely pass unsupported funtions when building ValueSpecification from Rawlambda
+      try {
+        const valueSpec =
+          this.editorStore.graphManagerState.graphManager.buildValueSpecification(
+            this.editorStore.graphManagerState.graphManager.serializeRawValueSpecification(
+              query,
+            ),
+            this.editorStore.graphManagerState.graph,
+          );
+
+        if (valueSpec instanceof LambdaFunctionInstanceValue) {
+          return this.getBindingWithParamRecursively(
+            valueSpec.values[0]?.expressionSequence[0],
+          );
+        }
+      } catch (error) {
+        this.editorStore.applicationStore.logService.error(
+          LogEvent.create(
+            LEGEND_STUDIO_APP_EVENT.TEST_DATA_GENERATION__FAILURE,
```

Comment:
should this really be logged as test data generation failure ?

Reply:
Created a new entry  `TEST_DATA_GENERATION__SETUP__FAILURE = 'editor.test.test-data-generation-setup.failure'` 
Let me know if this is better

Reply:
my point was that the function as nothing to do with test data generation. It is how you define the service test parameters values. So like test setup maybe but nothing to do with generation.